openc3-cosmos-script-runner-api/app/channels/application_cable/connection.rb

```diff
@@ -12,15 +12,36 @@
 # but WITHOUT ANY WARRANTY; without even the implied warranty of
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 # GNU Affero General Public License for more details.
-
-# Modified by OpenC3, Inc.
-# All changes Copyright 2022, OpenC3, Inc.
-# All Rights Reserved
 #
-# This file may also be used under the terms of a commercial license 
+# This file may also be used under the terms of a commercial license
 # if purchased from OpenC3, Inc.
 
+require 'openc3/utilities/authorization'
+
 module ApplicationCable
   class Connection < ActionCable::Connection::Base
+    include OpenC3::Authorization
+
+    identified_by :uuid
+    identified_by :scope
+
+    def connect
+      authorization('script_view')
+      self.uuid = SecureRandom.uuid
+      self.scope = request.headers['HTTP_X_OPENC3_SCOPE'] || request.query_parameters[:scope]
+    end
+
+    def authorization(permission)
+      begin
+        authorize(
+          permission: permission,
+          scope: request.headers['HTTP_X_OPENC3_SCOPE'] || request.query_parameters[:scope],
+          token: request.headers['HTTP_AUTHORIZATION'] || request.query_parameters[:authorization],
```

Comment:
Who sets these headers? Or are they always in the query_parameters?

Reply:
Always query parameters.  Will update.